alexandria/core/views.py

```diff
@@ -119,8 +120,16 @@ def _write_zip(self, file_obj, queryset):
                     temp_file.write(d)
 
                 temp_file.seek(0)
+
+                name = file.name
+                if file.name in seen_names:
+                    name_parts = name.split(".")
+                    name_parts[-2] += f"({seen_names.get(file.name)})"
+                    name = ".".join(name_parts)
+                seen_names[file.name] = seen_names.get(file.name, 0) + 1
+
```

Comment:
This feels a bit fragile (see below).  In another project with a similar problem, I've used something along those lines:

```suggestion

                name = file.name
                suffixes = itertools.count(start=1)
                while name in seen_names:
                    base_name, extension = file.name.rsplit('.', 1)
                    name = f"{base_name}({next(suffixes)}).{extension}"
                seen_names.add(name) # use a set instead of a dict for seen_names

```

For example, if there are the following files, you'd get a nasty surprise:

* foo.txt
* foo.txt
* foo(1).txt

In this case, in the second iteration, foo.txt would be (correctly) renamed to foo(1).txt, but in the third, you wouldn't notice the duplicate.

Reply:
Implemented and added test for that case